### PR TITLE
Support ``autodoc_mock_imports``

### PR DESCRIPTION
**Pull request by prajeeshag**
Sphinx autodoc_mock_imports is implemented here. 
https://github.com/ashb/sphinx-argparse/issues/9#issue-1097057823

### CODE REVIEW

**Review by ashb — CHANGES_REQUESTED** [review]
Please add some unit tests for this.

File: sphinxarg/ext.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
 from docutils.statemachine import StringList
 from docutils.utils import new_document
 from sphinx.util.nodes import nested_parse_with_titles
+from sphinx.ext.autodoc import mock
```
Comment (ashb) [inline sphinxarg/ext.py:13]: Could this import ever fail? I forget if autodoc is shipped with Sphinx or an extra module that needs to be installed.

**Comment by mark-nicholson** [comment]
Something is awry with this commit.  I get a crash at that first line.  version 0.4.0 work like a champ for a long time.
```
  File "/u02/masatest/mjnichol-3.6.1.dev1-bscrub-311/venv/lib64/python3.11/site-packages/sphinxarg/ext.py", line 509, in run
    with mock(self.config.autodoc_mock_imports):
              ^^^^^^^^^^^
AttributeError: 'ArgParseDirective' object has no attribute 'config'
```

**Comment by AA-Turner** [comment]
Fixed in 11bfc2bbc0fec600acd1d59e3cfa4d8e92379eb7; sorry. See also #56.

A

**Pull request merged**